SilKit/source/services/orchestration/TimeSyncService.cpp

```diff
@@ -475,10 +478,15 @@ void TimeSyncService::ExecuteSimStep(std::chrono::nanoseconds timePoint, std::ch
 {
     SILKIT_ASSERT(_simTask);
     using DoubleMSecs = std::chrono::duration<double, std::milli>;
+    using DoubleSecs = std::chrono::duration<double>;
 
     _waitTimeMonitor.StopMeasurement();
-    Trace(_logger, "Starting next Simulation Task. Waiting time was: {}ms",
-          std::chrono::duration_cast<DoubleMSecs>(_waitTimeMonitor.CurrentDuration()).count());
+    const auto waitingDuration = _waitTimeMonitor.CurrentDuration();
```

Comment:
This gets called before the first call to `_waitTimeMonitor.StartMeasurement();`. So the first waitingDuration will be max

Reply:
The first sample is now skipped. Thanks! :+1: